### ISSUE TITLE
Build error on openSUSE Leap 15.3 using gcc 10

### ISSUE THREAD

**Issue by coffeinflash** [peak3d/inputstream.adaptive#859]
Trying to build the latest master on openSUSE Leap 15.3 fails with:

```
  111s] cd "/home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/build/wvdecrypter" && /usr/bin/g++-10 -DBUILD_KODI_ADDON -DUNICODE -DWIDEVINECDMFILENAME=\"libwidevinecdm.so\" -D_UNICODE -D__STDC_FORMAT_MACROS -Dssd_wv_EXPORTS -I"/home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/lib/webm_parser/include" -I"/home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/cdm" -fmessage-length=0 -grecord-gcc-switches -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -pthread -DNDEBUG -flto -O2 -g -DNDEBUG -fPIC   -DTARGET_POSIX -DTARGET_LINUX -D_GNU_SOURCE -DHAVE_LINUX_UDMABUF=1 -DHAVE_LINUX_DMA_BUF=1 -DHAVE_MKOSTEMP=1 -DHAVE_LINUX_MEMFD=1 -DHAVE_STATX=1 -std=c++17 -MD -MT wvdecrypter/CMakeFiles/ssd_wv.dir/wvdecrypter.cpp.o -MF CMakeFiles/ssd_wv.dir/wvdecrypter.cpp.o.d -o CMakeFiles/ssd_wv.dir/wvdecrypter.cpp.o -c "/home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp"
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:176:23: error: 'virtual const char* WV_CencSingleSampleDecrypter::GetSessionId()' marked 'override', but does not override
[  111s]   176 |   virtual const char *GetSessionId() override;
[  111s]       |                       ^~~~~~~~~~~~
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:193:22: error: 'virtual AP4_Result WV_CencSingleSampleDecrypter::SetFragmentInfo(AP4_UI32, const AP4_UI08*, AP4_UI08, AP4_DataBuffer&, AP4_UI32)' marked 'override', but does not override
[  111s]   193 |   virtual AP4_Result SetFragmentInfo(AP4_UI32 pool_id, const AP4_UI08 *key, const AP4_UI08 nal_length_size,
[  111s]       |                      ^~~~~~~~~~~~~~~
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:195:20: error: 'virtual AP4_UI32 WV_CencSingleSampleDecrypter::AddPool()' marked 'override', but does not override
[  111s]   195 |   virtual AP4_UI32 AddPool() override;
[  111s]       |                    ^~~~~~~
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:196:16: error: 'virtual void WV_CencSingleSampleDecrypter::RemovePool(AP4_UI32)' marked 'override', but does not override
[  111s]   196 |   virtual void RemovePool(AP4_UI32 poolid) override;
[  111s]       |                ^~~~~~~~~~
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:199:22: error: 'virtual AP4_Result WV_CencSingleSampleDecrypter::DecryptSampleData(AP4_UI32, AP4_DataBuffer&, AP4_DataBuffer&, const AP4_UI08*, unsigned int, const AP4_UI16*, const AP4_UI32*)' marked 'override', but does not override
[  111s]   199 |   virtual AP4_Result DecryptSampleData(AP4_UI32 pool_id,
[  111s]       |                      ^~~~~~~~~~~~~~~~~
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp: In constructor 'WV_CencSingleSampleDecrypter::WV_CencSingleSampleDecrypter(WV_DRM&, AP4_DataBuffer&, const uint8_t*, bool)':
[  111s] /home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/wvdecrypter/wvdecrypter.cpp:422:3: error: 'SetParentIsOwner' was not declared in this scope
[  111s]   422 |   SetParentIsOwner(false);
[  111s]       |   ^~~~~~~~~~~~~~~~
[  111s] make[2]: *** [wvdecrypter/CMakeFiles/ssd_wv.dir/build.make:79: wvdecrypter/CMakeFiles/ssd_wv.dir/wvdecrypter.cpp.o] Error 1
[  111s] make[2]: Leaving directory '/home/abuild/rpmbuild/BUILD/kodi.binary-addons-inputstream.adaptive-20.0.3.20211210T124045~0b2715a/build'
[  111s] make[1]: *** [CMakeFiles/Makefile2:263: wvdecrypter/CMakeFiles/ssd_wv.dir/all] Error 2
[  111s] make[1]: *** Waiting for unfinished jobs....
```

The complete log can be found here at [OBS](https://build.opensuse.org/package/live_build_log/home:Herbster0815:HTPC/kodi.binary-addons-inputstream.adaptive/openSUSE_Leap_15.3/x86_64)

The package sources can be found here at [OBS](https://build.opensuse.org/package/show/home:Herbster0815:HTPC/kodi.binary-addons-inputstream.adaptive)

**Comment by coffeinflash** [comment]
This is related to my external bento4-package. So this can be closed.

**Comment by coffeinflash** [comment]
Fixed the bento4 package. Now it built successfully.

**Comment by coffeinflash** [comment]
solved - no bug.

**Comment by bassmadrigal** [comment]
@coffeinflash, can you provide the fix? I'm running into the same issue with a bento4 package built from the latest bento4 commit.

https://xkcd.com/979/ ;)

If I attempt to build [xbmc/Bento4](https://github,com/xbmc/Bento4) with my build script, it is unable to finish unless I disable the apps (with apps disabled, it builds fine).

```[ 79%] Building CXX object CMakeFiles/mp42avc.dir/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp.o
In file included from /tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4.h:230,
                 from /tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp:35:
/tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp: In function ‘void DecryptAndWriteSamples(AP4_Track*, AP4_SampleDescription*, AP4_Byte*, AP4_ByteStream*)’:
/tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp:231:52: error: no matching function for call to ‘AP4_SampleDecrypter::DecryptSampleData(AP4_DataBuffer&, AP4_DataBuffer&)’
  231 |         if (AP4_FAILED(decrypter->DecryptSampleData(encrypted_data, decrypted_data))) {
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4Results.h:67:30: note: in definition of macro ‘AP4_FAILED’
   67 | #define AP4_FAILED(result) ((result) != AP4_SUCCESS)
      |                              ^~~~~~
In file included from /tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4OmaDcf.h:41,
                 from /tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4.h:285,
                 from /tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp:35:
/tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4Protection.h:396:24: note: candidate: ‘virtual AP4_Result AP4_SampleDecrypter::DecryptSampleData(AP4_UI32, AP4_DataBuffer&, AP4_DataBuffer&, const AP4_UI08*)’
  396 |     virtual AP4_Result DecryptSampleData(AP4_UI32 poolid,
      |                        ^~~~~~~~~~~~~~~~~
/tmp/SBo/Bento4-1.6.0-639-6-Nexus/Source/C++/Core/Ap4Protection.h:396:24: note:   candidate expects 4 arguments, 2 provided
make[2]: *** [CMakeFiles/mp42avc.dir/build.make:76: CMakeFiles/mp42avc.dir/Source/C++/Apps/Mp42Avc/Mp42Avc.cpp.o] Error 1
```

If I use the original [axiomatic-systems/Bento4](https://github,com/axiomatic-systems/Bento4) source with apps enabled, inputstream.adaptive errors out with the same error above.

@CastagnaIT, Is it possible to build this with an external bento4 with the apps enabled? If it isn't working right now, that's fine and I can just use the internal build of it, but, if possible, I prefer to support external programs when practical to minimize overall system resources.

Thanks for your time and effort involved with kodi and these addons!

**Comment by CastagnaIT** [comment]
you cant use upstream source to build ISA otherwise you need to apply to upstream our commits changes, at this point we might as well use our fork

we dont provide support to fix bento apps, are not required to our ISA project
